[Denis] Duke Increments

WeomuCat opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

Commit:
Merge branch 'gradle'

Commit:
Added gradle.

Commit:
Added greet, echo, bye.

Commit:
Added add, list.

Commit:
Added Task class.

Commit:
Added done.

Commit:
Added task inheritance.

Commit:
Added todo, event, deadline.

Commit:
Added text-ui-test.

Commit:
Rename Tasks.

Commit:
Added exceptions.

Commit:
Added error handling.

Commit:
Added delete.

Commit:
Convert spaces to tabs.

Commit:
Added the ability to save to disk.

Commit:
Teach Duke about the concept of Dates and Times.

Commit:
Merge branch 'branch-Level-8'

Commit:
Make the code more OOP.

Commit:
Divide classes into packages.

Commit:
Add JUnit to gradle.

Commit:
Improve Parser to preserve whitespace.

Commit:
Bug fixes.

- Fix bug where using wrong listeners when loading from storage.
- Refactor TaskList.
- Added InvalidTaskIndexException.

Commit:
Add JUnit tests.

Commit:
Configure gradle for packaging Jar.

Commit:
Use Google Java Styleguide.

- https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml

Commit:
Added JavaDoc style documentation.

Commit:
Added find command.

- JUnit not done.

Commit:
Added FindTask case insensitivity.

Commit:
Added FindTask JUnit tests.

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-Level-9'

Commit:
Added checkstyle.

Using https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml

waifonglee reviewed (commented):
Nicely structured code that follows coding standard :). Good job packaging code to make it more structured.

arjunwangty commented on src/main/java/weomucat/duke/Controller.java:
```diff
@@ -0,0 +1,211 @@
+
+  /**
+   * Add a AddTaskCommandListener to the Controller. When a AddTaskCommand is received, this
+   * listener will be notified.
```
Javadoc is detailed and informative 👍One minor suggestion is to separate the first sentence as a summary of the method and start the sentence with an action word in the form "Adds", "Parses, Performs" etc. :)

WeomuCat replied:
Hi Tianyi, thank you for the kind words :) I agree with the javadocs, I will take your advice and put a summary with an action word above my javadocs! 👍

arjunwangty reviewed (commented):
Hi Denis, I really like the rigour of your code. Your classes are very well encapsulated, commands classes and listeners classes are also extracted out. I think I can learn how to write such well-designed codes :) I can see that your project is also properly packaged and built with Gradle. Well done!

Commit:
Added GUI to Duke!

Commit:
Merge branch 'branch-Level-10'

Commit:
Fixed checkstyle issues.

- Fix GUI not using correct encoding.

Commit:
Add shadow to gradle for javafx archive.

Commit:
Increment version to 0.2.

Commit:
Add assertions.

Commit:
Merge pull request #3 from WeomuCat/branch-A-Assertions

Add assertions.

Commit:
Refactor GUI Messages.

- Set proper access modifiers.
- Improved Javadoc comments.

Commit:
Add .travis.yml.

Commit:
Make gradlew executable for travis.

Commit:
Add skip travis build if is a tag.

Commit:
Configure travis to build on multiple os.

Commit:
Remove windows from travis because it does not support building java.

Commit:
Added Date & Event classes for EventTask.

Commit:
Refactor Message to MessageBox.

- Added title for messages
- Added DateRange

Commit:
Added event_at command.

- For selecting tentatively scheduled slots in events.
- Refactored TaskListeners.
- TODO: Storage

Commit:
Fix Storage.

Commit:
Fix javafx linux issue.

Commit:
Added task message colors & shadows.

Commit:
Added snooze command for snoozable tasks like deadines.

Commit:
Added user input history to GraphicalUi.

- Use arrow keys to scroll through history.
- Fixed typo for snooze command.

Commit:
Added framework of recurring tasks.

- Add more parsers
- Refactor commands & date

Commit:
Added framework of recurring tasks.

- Add more parsers
- Refactor commands & date

Commit:
Added NumberedTaskLists.

Commit:
Added NumberedTaskLists.

Commit:
[skip travis] Still figuring out how to implement recurring tasks.

Commit:
[skip travis] Still figuring out how to implement recurring tasks.

Commit:
Added recurring tasks.

- Refactor Controller & Commands.

Commit:
Added recurring tasks.

- Refactor Controller & Commands.

Commit:
Merge branch 'branch-B-RecurringTasks' into develop

Commit:
Fix tests & checkstyle.

Commit:
Merge branch 'branch-B-RecurringTasks' into develop

Commit:
Added command parameter options.

Commit:
Add more JUnit tests.

- Refactor parameters & dates.

Commit:
Added HeterogeneousContainers.

- Quite proud of this creation!
- Continuing to refactor code.

Commit:
Add container to TaskManager.

Commit:
Added Help Command.

Commit:
Refactor Messages.

Commit:
Added MessageContent & MessageBoxOptions.

Commit:
Added cross platform support.

Commit:
Update User Guide.

Commit:
Set theme jekyll-theme-merlot

Commit:
Set theme jekyll-theme-architect

Commit:
Set theme jekyll-theme-minimal

Commit:
Set theme jekyll-theme-slate

Commit:
Update User Guide.

Commit:
Update User Guide.

Commit:
Remove theme.

Commit:
Update User Guide.

Commit:
Set theme jekyll-theme-cayman

Commit:
Update User Guide.

Commit:
Update User Guide.

Commit:
Update User Guide.

Commit:
Merge branch 'master' into develop

Commit:
Release 0.3.

Commit:
Merge branch 'master' of https://github.com/WeomuCat/duke

Commit:
Update User Guide.

Commit:
Rename Duke to Doko.

Commit:
Change tasks file path.

Commit:
Added custom tasks on first startup.

Commit:
Added User Guide media.

Commit:
Merge branch 'master' into develop

Commit:
Merge branch 'master' into develop

Commit:
Added media to User Guide.

Commit:
Fix User Guide media links.

Commit:
Fix User Guide media links.
